embedded/store/immustore.go

```diff
@@ -805,8 +805,21 @@ func (s *ImmuStore) Snapshot() (*Snapshot, error) {
 	}, nil
 }
 
-func (s *ImmuStore) SnapshotSince(tx uint64) (*Snapshot, error) {
-	snap, err := s.indexer.SnapshotSince(tx)
+func (s *ImmuStore) SnapshotMustIncludeTxID(snapshotMustIncludeTxID uint64) (*Snapshot, error) {
+	return s.SnapshotMustIncludeTxIDWithRenewalPeriod(snapshotMustIncludeTxID, 0)
+}
+
+func (s *ImmuStore) SnapshotMustIncludeTxIDWithRenewalPeriod(snapshotMustIncludeTxID uint64, snapshotRenewalPeriod time.Duration) (*Snapshot, error) {
```

Comment:
It would be very helpful to understand this if you could add a comment on what exactly is happening as the func name is verbose enough, but finding it bit hard to grasp